demo/video_gpuaccel_demo.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+
+import cv2
+import mmcv
+import torch
+import tqdm
+import numpy as np
+from torchvision.transforms import functional as F
+from mmdet.apis import init_detector
+from mmdet.datasets.pipelines import Compose
+
+try:
+    import ffmpegcv
+except ImportError:
+    raise ImportError('Please install ffmpegcv with:\n\n    pip install ffmpegcv')
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='MMDetection video demo')
+    parser.add_argument('video', help='Video file')
+    parser.add_argument('config', help='Config file')
+    parser.add_argument('checkpoint', help='Checkpoint file')
+    parser.add_argument(
+        '--device', default='cuda:0', help='Device used for inference')
+    parser.add_argument(
+        '--score-thr', type=float, default=0.3, help='Bbox score threshold')
+    parser.add_argument('--out', type=str, help='Output video file')
+    parser.add_argument('--show', action='store_true', help='Show video')
+    parser.add_argument('--nvdecode', action='store_true', help='Use NVIDIA decorder')
+    parser.add_argument(
+        '--wait-time',
+        type=float,
+        default=1,
+        help='The interval of show (s), 0 is block')
+    args = parser.parse_args()
+    return args
+
+
+def prefetch_img_metas(model, ori_wh):
+    cfg = model.cfg
+    w, h = ori_wh
+    cfg.data.test.pipeline[0].type = 'LoadImageFromWebcam'
+    test_pipeline = Compose(cfg.data.test.pipeline)
+    data = {'img':np.zeros((h, w, 3), dtype=np.uint8)}
+    data = test_pipeline(data)
+    img_metas = data['img_metas'][0].data
+    return img_metas
+
+
+def process_img(frame_resize, img_metas):
+    assert frame_resize.shape==img_metas['pad_shape']
+    frame_cuda = torch.from_numpy(frame_resize.astype(np.float32)).cuda()
```

Comment:
No need to convert to float, it must be float when norm is calculated.

`frame_cuda = torch.from_numpy(frame_resize).cuda()`

Reply:
The F.normalize need the float data type. 
```python
frame_cuda = torch.from_numpy(frame_resize).cuda()
frame_cuda = F.normalize(frame_cuda, mean=mean, std=std)
```
>Traceback (most recent call last):
  File "demo/video_gpuaccel_demo.py", line 103, in <module>
    main()
  File "demo/video_gpuaccel_demo.py", line 85, in main
    data = process_img(frame_resize, img_metas)
  File "demo/video_gpuaccel_demo.py", line 56, in process_img
    frame_cuda = F.normalize(frame_cuda, mean=mean, std=std)
  File "/home/liying_lab/chenxinfeng/.conda/envs/cbnet/lib/python3.7/site-packages/torchvision/transforms/functional.py", line 298, in normalize
    tensor.sub_(mean).div_(std)
RuntimeError: result type Float can't be cast to the desired output type Byte

I would keep the `frame_cuda = torch.from_numpy(frame_resize.astype(np.float32)).cuda()`

Reply:
Then you can replace it with the following
`frame_cuda = torch.from_numpy(frame_resize).float().cuda()`

Reply:
And. Is it possible for this program to run on CPU？

Reply:
OK, I will swich to ` torch.from_numpy(a).cuda().float()`
```
    b = torch.from_numpy(a).float().cuda()     # slow; CPU 2,000%
    d = torch.from_numpy(a.astype(numpy.float32)).cuda() # fast; CPU 100%
    c = torch.from_numpy(a).cuda().float()    # fastest; CPU 100%
```

Reply:
This program can run in CPU by replacing the '.cuda() ' to '.to(device)' .

Reply:
en. So it's better to change it to be more general.

Reply:
> en. So it's better to change it to be more general.

This script is called "gpuaccel_demo", so I think cuda only is acceptable.